### DISCUSSION TITLE
JSON.parse() - How to handle throwable error in safeParse?

### DISCUSSION THREAD

**Discussion by enheit** [open-circle/valibot#537] [Q&A]
I have such schema

```js
export const Schema = object({
  nextCursor: optional(
    transform(string(), (input) => { 
      try { 
        return JSON.parse(input) as Record<string, string> 
      } catch { 
        // What to return here? Error? ValidError? String?
      } 
    },
  ))
})
```
And use it like this
```ts
const nextKey = "{ \"hasNext\": \"true\" }";
const result = v.safeParse(Schema, { nextKey });
```

If `JSON.parse(...)` throws an error, how can I notify a `Valibot` that this error is expected and just return a message like "JSON parse failed" inside `result.issues`? Currently (without a `try`/`catch`), it throws an error, and `safeParse` doesn't handle it.

**Comment by fabian-hiller** [comment]
Unfortunately, we do not have a perfect answer to this. But you could try to return the input and use the third argument of `transform` to validate its output.

**Reply by enheit** [reply]
Could you provide a little example of how to do this? I don't quite understand what this means

> use the third argument of `transform` to validate its output

Thanks

**Reply by fabian-hiller** [reply]
```ts
const Schema = v.object({
  nextCursor: v.optional(
    v.transform(
      v.string(),
      (input) => {
        try {
          return JSON.parse(input) as Record<string, string>;
        } catch {
          return null;
        }
      },
      v.record(v.string(), 'Optional errer message here') // <--
    )
  ),
});
```
